### PR TITLE
TRUNK-5773: Migrate From org.glassfish.jaxb to com.sun.xml.bind

### PR DESCRIPTION
**Pull request by achilep**
## Description of what I changed

This Sun-task is to migrate  Migrate From org.glassfish.jaxb to com.sun.xml.bind  this imply this change:
org.glassfish.jaxb:jaxb-runtime:2.3.2 --> com.sun.xml.bind:jaxb-impl:3.0.0-M4

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5773

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -551,7 +551,7 @@
 			<dependency>
 			    <groupId>org.glassfish.jaxb</groupId>
 			    <artifactId>jaxb-runtime</artifactId>
-			    <version>2.3.2</version>
+			    <version>3.0.0-M4</version>
```
Comment (HerbertYiga) [inline pom.xml:554]: why upgrading to -M4?

Comment (achilep) [reply pom.xml:554]: thanks @HerbertYiga  [here](https://search.maven.org/artifact/org.glassfish.jaxb/jaxb-runtime)  is the maven repository and from that page we can see that 3.0.0-M is the last version that the have .They have not yet release 3.0.0 version.

**Comment by dkayiwa** [comment]
https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by achilep** [comment]
Screenshot from 2020-06-19 21-16-29

**Comment by achilep** [comment]
hello @dkayiwa  @mozzy11

**Comment by dkayiwa** [comment]
@achilep what is the reason for this migration?

**Comment by achilep** [comment]
@ibacher and @mozzy11  suggests it because the old ones: org.glassfish.jaxb. had a problem updating it. so they suggested that we use this one instead.

@dkayiwa , here's the reason they gave:

> See [here](https://eclipse-ee4j.github.io/jaxb-ri/). Incidentally, we might see if things work using the com.sun.xml.bind:jaxb-impl artifact named there as that, rather than the Glassfish implementation, appears to be the reference implementation for JAX-B now.

**Comment by dkayiwa** [comment]
Since this is not just a version upgrade, the ticket description needs to explicitly state why are replacing this library.

**Pull request merged**